src/veh_interact.cpp

```diff
@@ -71,7 +71,11 @@ player_activity veh_interact::run( vehicle &veh, int x, int y )
             time = vp->install_time( g->u );
             break;
         case 'r':
-            time = pt->is_broken() ? vp->install_time( g->u ) : ( vp->repair_time( g->u ) * pt->base.damage() );
+            if( pt->is_broken() ) {
+                time = vp->install_time( g->u );
+            } else {
+                time = vp->repair_time( g->u ) * double( pt->base.damage() ) / pt->base.max_damage();
```

Comment:
> the repair code block is never called!

Yes, it's assumed, but isn't guaranteed as the code evolves. I suppose this assertion deserves either an `assert` / clause / comment here, in `veh_interact::run()`. BTW `int vehicle_part::hp()` (called by `is_broken()`) has the very same issue already.